### PR TITLE
Implement Interop intrinsics as listed in truffle.h

### PR DESCRIPTION
**Pull request by grimmerm**
truffle.h defines signatures for interop functions that allow Sulong to access foreign objects (see #157). The following change provides an implementation for these functions. The functions are implemented as Sulong runtime intrinsics.

### CODE REVIEW

File: mx.sulong/mx_sulong.py [hunk 1]
```diff
@@ -82,6 +85,9 @@ def travis1(args=None):
         with Task('TestPolglot', tasks) as t:
             if t: runPolyglotTestCases()
     with VM('server', 'product'):
+        with Task('TestPolglot', tasks) as t:
```
Comment (mrigger) [inline mx.sulong/mx_sulong.py]: TestInterop?

File: mx.sulong/mx_sulong.py [hunk 2]
```diff
@@ -468,6 +475,11 @@ def runPolyglotTestCases(args=None):
     vmArgs, _ = truffle_extract_VM_args(args)
     return unittest(getCommonUnitTestOptions() + vmArgs + ['com.oracle.truffle.llvm.test.TestPolyglotEngine'])
 
+def runInteropTestCases(args=None):
+    """runs the type test cases"""
```
Comment (mrigger) [inline mx.sulong/mx_sulong.py]: runs the interop test cases?

File: mx.sulong/mx_sulong.py [hunk 3]
```diff
@@ -56,6 +56,9 @@ def executeGate():
         with Task('TestPolglot', tasks) as t:
```
Comment (eregon) [inline mx.sulong/mx_sulong.py]: here too.

File: include/truffle.h [hunk 4]
```diff
@@ -76,26 +81,24 @@ char truffle_unbox_c(void *object);
...
+double 	truffle_read_d 	(void *object, const char *name);
+bool 	truffle_read_b 	(void *object, const char *name);
+
+void *	truffle_read_idx 	(void *object, int idx);
+int 	truffle_read_idx_i 	(void *object, int idx);
+long 	truffle_read_idx_l 	(void *object, int idx);
+char 	truffle_read_idx_c 	(void *object, int idx);
+float 	truffle_read_idx_f 	(void *object, int idx);
+double 	truffle_read_idx_d 	(void *object, int idx);
+bool 	truffle_read_idx_b 	(void *object, int idx);
```
Comment (woess) [inline include/truffle.h]: I don't like tabs in source files.

File: mx.sulong/mx_sulong.py (hunk 3 above)
Comment (eregon) [inline mx.sulong/mx_sulong.py]: (Above) `dists/sulong-mx.jar` should not be committed I think.

File: include/truffle.h (hunk 4 above)
Comment (mrigger) [inline include/truffle.h]: Can you also add a short description of the motivation for the changes in the PR?

File: mx.sulong/mx_sulong.py (hunk 3 above)
Comment (woess) [inline mx.sulong/mx_sulong.py]: @mrigger That could be another think to check for your gitlogcheck.

File: projects/com.oracle.truffle.llvm.test/tests/interop/interop004.c [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	sum+= truffle_read_idx_l 	(l, 0);
+	sum+= truffle_read_idx_c 	(c, 0);
+	sum+= truffle_read_idx_f 	(f, 0);
+	sum+= truffle_read_idx_d 	(d, 0);
+	
+	sum+= truffle_read_idx_i 	(i, 1);
+	sum+= truffle_read_idx_l 	(l, 1);
+	sum+= truffle_read_idx_c 	(c, 1);
+	sum+= truffle_read_idx_f 	(f, 1);
+	sum+= truffle_read_idx_d 	(d, 1);
```
Comment (woess) [inline projects/com.oracle.truffle.llvm.test/tests/interop/interop004.c]: formatting

File: include/truffle.h (hunk 4 above)
Comment (mrigger) [inline include/truffle.h]: @woess #177 introduces a format check of the file.

File: mx.sulong/mx_sulong.py (hunk 3 above)
Comment (mrigger) [inline mx.sulong/mx_sulong.py]: @woess I think that an entry in the `.gitignore` file is sufficient for now: #179.

**Pull request merged**